STYLE.md

```diff
@@ -0,0 +1,217 @@
+# Introduction
+
+This file describes the various style and design conventions that are used in the Backstage main repository. While you may choose to use these conventions in your own Backstage projects, it is not required.
+
+# TypeScript Coding Conventions
+
+Our TypeScript style is inspired by the [style guidelines](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines) of the TypeScript implementation itself.
+
+## Naming
+
+1. Use PascalCase for type names.
+1. Do not use `I` as a prefix for interface names.
+1. Use PascalCase for `enum` values.
+1. Use camelCase for function names.
+1. Use camelCase for property names and local variables.
+1. Do not use `_` as a prefix for private properties.
```

Comment:
Does use `#` work as a good recommendation here?

Reply:
Not at the moment, not supported by storybook yet ._.

Reply:
Ah ok - I thought I saw a PR fixing that, but maybe it was just removing them :joy: